fugue-deploy-aws/src/main/java/com/symphony/oss/fugue/aws/deploy/AwsFugueDeploy.java

```diff
@@ -1762,17 +1766,32 @@ protected void postDeployLambdaContainer(String name, Collection<String> paths,
         if(!paths.isEmpty())
           createApiGatewayPaths(getFunctionInvocationArn(functionName), IntegrationType.AWS_PROXY, POST, null, null, paths);
         
-        if(!subscriptions.isEmpty())
-          createLambdaSubscriptions(functionName, subscriptions);
+         createLambdaSubscriptions(functionName, subscriptions);
       }
     }
 
     private void createLambdaSubscriptions(String functionName, Collection<Subscription> subscriptions)
     {
+      HashMap<String, EventSourceMappingConfiguration> mappings = new HashMap<>();
+      
+      ListEventSourceMappingsResult mappingResult = lambdaClient_.listEventSourceMappings(new ListEventSourceMappingsRequest()
+          .withFunctionName(functionName)
+          );
+      
+      for(EventSourceMappingConfiguration mapping : mappingResult.getEventSourceMappings())
+        mappings.put(mapping.getEventSourceArn(), mapping);
+      
       for(Subscription subscription : subscriptions)
       {
+        if(mappings.containsKey(subscription.getSource()))
+          mappings.remove(subscription.getSource());
```

Comment:
remove(key) returns the previous mapping so you could have just done 

if(remove(subscription.getSource()) != null)